lazy.go

```diff
@@ -51,7 +51,11 @@ type lazyConn struct {
 }
 
 func (l *lazyConn) Read(b []byte) (int, error) {
-	if !l.rhandshake {
+	l.rhlock.RLock()
+	rhandshake := l.rhandshake
+	l.rhlock.RUnlock()
```

Comment:
I agree that it might be only case when races are "good" but it screws over race test in all code that uses this. This means that we can't run race detector on big part of go-libp2p and go-ipfs code unless we tolerate false positives.